src/libraries/System.Configuration.ConfigurationManager/src/System/Configuration/ClientConfigPaths.cs

```diff
@@ -47,8 +48,11 @@ private ClientConfigPaths(string exePath, bool includeUserConfig)
             {
                 // Exe path wasn't specified, get it from the entry assembly
                 exeAssembly = Assembly.GetEntryAssembly();
+                var exeName = exeAssembly?.ManifestModule.Name;
```

Comment:
It would be better to use `exeAssembly!.Location`. It is simpler and it returns empty string for files loaded from memory. Testing for empty string looks better than testing for magic `<Unknown>` string.

Reply:
Agreed, it is better. I was trying to find something in `Environment` related to single file vs. other types of hosting for robust detection, but there wasn't one, so had to improvise. 😁